pkg/vault/test/foundry/utils/BaseMedusaTest.sol

```diff
@@ -0,0 +1,223 @@
+import { IVaultMock } from "@balancer-labs/v3-interfaces/contracts/test/IVaultMock.sol";
+import { IBasePool } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePool.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import { IStdMedusaCheats } from "@balancer-labs/v3-interfaces/contracts/test/IStdMedusaCheats.sol";
+import "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { InputHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/InputHelpers.sol";
+import { CREATE3 } from "@balancer-labs/v3-solidity-utils/contracts/solmate/CREATE3.sol";
+import { ERC20TestToken } from "@balancer-labs/v3-solidity-utils/contracts/test/ERC20TestToken.sol";
+import { WETHTestToken } from "@balancer-labs/v3-solidity-utils/contracts/test/WETHTestToken.sol";
+
+import { VaultExtensionMock } from "../../../contracts/test/VaultExtensionMock.sol";
+import { VaultAdminMock } from "../../../contracts/test/VaultAdminMock.sol";
+import { VaultMock } from "../../../contracts/test/VaultMock.sol";
+import { ProtocolFeeController } from "../../../contracts/ProtocolFeeController.sol";
+import { BasicAuthorizerMock } from "../../../contracts/test/BasicAuthorizerMock.sol";
+import { RouterMock } from "../../../contracts/test/RouterMock.sol";
+import { BatchRouterMock } from "../../../contracts/test/BatchRouterMock.sol";
+import { CompositeLiquidityRouterMock } from "../../../contracts/test/CompositeLiquidityRouterMock.sol";
+import { ProtocolFeeControllerMock } from "../../../contracts/test/ProtocolFeeControllerMock.sol";
+import { PoolFactoryMock } from "../../../contracts/test/PoolFactoryMock.sol";
+
+contract BaseMedusaTest is Test {
+    IStdMedusaCheats internal medusa = IStdMedusaCheats(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+
+    event Debug(string, uint256);
+
+    IPermit2 internal permit2;
+
+    // Main contract mocks.
+    IVaultMock internal vault;
+    IVaultExtension internal vaultExtension;
+    IVaultAdmin internal vaultAdmin;
+    RouterMock internal router;
+    BatchRouterMock internal batchRouter;
+    CompositeLiquidityRouterMock internal compositeLiquidityRouter;
+    BasicAuthorizerMock internal authorizer;
+    ProtocolFeeControllerMock internal feeController;
+    PoolFactoryMock internal factoryMock;
+
+    IBasePool internal pool;
+    uint256 internal poolCreationNonce;
+
+    uint256 internal constant DEFAULT_USER_BALANCE = 1e18 * 1e18;
+
+    // Set alice,bob and lp to addresses of medusa.json "senderAddresses" property
+    address internal alice = address(0x10000);
+    address internal bob = address(0x20000);
+    address internal lp = address(0x30000);
+
+    ERC20TestToken internal dai;
+    ERC20TestToken internal usdc;
+    WETHTestToken internal weth;
+
+    constructor() {
+        dai = _createERC20TestToken("DAI", "DAI", 18);
+        usdc = _createERC20TestToken("USDC", "USDC", 18);
+        weth = new WETHTestToken();
```

Comment:
If you don't care about native eth tests you can just make this a regular `ERC20TestToken`; it doesn't matter that it's called "WETH"